lms/djangoapps/instructor/views/api.py

```diff
@@ -2708,7 +2704,7 @@ def start_certificate_regeneration(request, course_id):
         )
 
     # Check if the selected statuses are allowed
-    allowed_statuses = GeneratedCertificate.get_unique_statuses(course_key=course_key, flat=True)
+    allowed_statuses = [CertificateStatuses.downloadable, CertificateStatuses.error, CertificateStatuses.notpassing]
```

Comment:
@saleem-latif I mean, sure... but that doesn't tell me anything the naming convention doesn't already communicate.
I'm asking, explicitly, what was the list of statuses?

With this change, this list is now:
- `downloadable`
- `error`
- `notpassing`

What was in this list before?

What, specifically, prompted this change?
I'm looking at the story [1] and nothing there requests that this list be changed from dynamic to static.

When @asadiqbal08 asked for an explanation, you instead repeated what the code is already telling us, hence my request for clarification.

The code always tells us _what_ is happening; you (and your commit messages) need to explain _why_ it was done..

[1] https://openedx.atlassian.net/browse/SOL-1418